src/test/java/teammates/test/cases/newaction/StudentGetCourseDetailsActionTest.java

```diff
@@ -0,0 +1,157 @@
+
+        StudentGetCourseDetailsResult output = (StudentGetCourseDetailsResult) r.getOutput();
+
+        StudentAttributes expectedStudent = output.getStudent();
+        expectedStudent.lastName = null;
+        expectedStudent.comments = null;
+        expectedStudent.key = null;
+        assertEquals(student1InCourse1.toString(), expectedStudent.toString());
+
+        assertEquals(student1InCourse1.course, output.getCourse().getId());
+
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        InstructorAttributes instructor2OfCourse1 = typicalBundle.instructors.get("instructor2OfCourse1");
+        InstructorAttributes instructor3OfCourse1 = typicalBundle.instructors.get("instructor3OfCourse1");
+        InstructorAttributes helperOfCourse1 = typicalBundle.instructors.get("helperOfCourse1");
+        InstructorAttributes instructorNotYetJoinCourse1 = typicalBundle.instructors.get("instructorNotYetJoinCourse1");
+
+        List<String> expectedInstructorNameList = new LinkedList<>();
+        expectedInstructorNameList.add(instructor1OfCourse1.getName());
+        expectedInstructorNameList.add(instructor2OfCourse1.getName());
+        expectedInstructorNameList.add(instructor3OfCourse1.getName());
+        expectedInstructorNameList.add(helperOfCourse1.getName());
+        expectedInstructorNameList.add(instructorNotYetJoinCourse1.getName());
+
+        AssertHelper.assertSameContentIgnoreOrder(expectedInstructorNameList, output.getInstructorNames());
+
+        StudentAttributes student2InCourse1 = typicalBundle.students.get("student2InCourse1");
+        StudentAttributes student3InCourse1 = typicalBundle.students.get("student3InCourse1");
+        StudentAttributes student4InCourse1 = typicalBundle.students.get("student4InCourse1");
+        StudentAttributes student5InCourse1 = typicalBundle.students.get("student5InCourse1");
+
+        List<StudentAttributes> expectedTeammatesList = new LinkedList<>();
+        expectedTeammatesList.add(student2InCourse1);
+        expectedTeammatesList.add(student3InCourse1);
+        expectedTeammatesList.add(student4InCourse1);
+        expectedTeammatesList.add(student5InCourse1);
+
+        List<StudentProfileAttributes> expectedTeammateProfiles = new LinkedList<>();
+        expectedTeammatesList.forEach(teammate -> {
+            StudentProfileAttributes teammateProfile = logic.getStudentProfile(teammate.googleId);
+            if (teammateProfile != null) {
+                teammateProfile.googleId = null;
+                teammateProfile.email = null;
+                teammateProfile.modifiedDate = null;
+
+                expectedTeammateProfiles.add(teammateProfile);
+            }
+        });
+
+        AssertHelper.assertSameContentIgnoreOrder(expectedTeammateProfiles, output.getTeammateProfiles());
+
+        ______TS("using a registered student in an archived course");
```

Comment:
I put a test case for archived course because in v6 the student can still access the archived course details, just in case in the future there is some different planning for the accessibility of archived courses by students.